### PR TITLE
Improve the API docs for the JSONAPI adapter

### PR DESCRIPTION
**Pull request by bmac**

### CODE REVIEW

File: addon/adapters/json-api.js [hunk 1]
```diff
@@ -9,6 +9,135 @@ import isEnabled from 'ember-data/-private/features';
 import { instrument, deprecate } from 'ember-data/-private/debug';
 
 /**
+  The `JSONAPIAdapter` is the default adapter used by Ember Data. It is
+  responsible for transforming the store's requests into HTTP request
```
Comment (locks) [inline addon/adapters/json-api.js]: HTTP requests

File: addon/adapters/json-api.js [hunk 2]
```diff
@@ -9,6 +9,135 @@ import isEnabled from 'ember-data/-private/features';
...
+  that follow the [JSON API](http://jsonapi.org/format/) format.
+
+  ## JSON API Conventions
+
+  The JSONAPIAdapter uses JSON API conventions for building the url
+  for a record and selecting the HTTP verb to use with a request.  The
```
Comment (locks) [inline addon/adapters/json-api.js]: Extra whitespace after period.

File: addon/adapters/json-api.js [hunk 3]
```diff
@@ -9,6 +9,135 @@ import isEnabled from 'ember-data/-private/features';
...
+    </th>
+    <td>
+      DELETE
+    </td>
+    <td>
+      /posts/123
+    </td>
+  </tr>
+</table>
+
```
Comment (locks) [inline addon/adapters/json-api.js]: Extra newline?

File: addon/adapters/json-api.js [hunk 4]
```diff
@@ -9,6 +9,135 @@ import isEnabled from 'ember-data/-private/features';
...
+  status code of the 2xx family ("Success"), as well as 304 ("Not
+  Modified"). Any other status code will be considered a failure.
+
+  On success, the request promise will be resolved with the full
+  response payload.
+
+  Failed responses with status code 422 ("Unprocessable Entity") will
+  be considered "invalid". The response will be discarded, except for
+  the `errors` key. The request promise will be rejected with a
+  `DS.InvalidError`.  This error object will encapsulate the saved
```
Comment (locks) [inline addon/adapters/json-api.js]: Extra whitespace after period.

File: addon/adapters/json-api.js [hunk 5]
```diff
@@ -9,6 +9,135 @@ import isEnabled from 'ember-data/-private/features';
...
+  `errors` value.
+
+  Any other status codes will be treated as an "adapter error". The
```
Comment (locks) [inline addon/adapters/json-api.js]: why the double quotes here and around `invalid`?

File: addon/adapters/json-api.js [hunk 6]
```diff
@@ -109,14 +246,6 @@ var JSONAPIAdapter = RESTAdapter.extend({
...
   coalesceFindRequests: false,
 
-  /**
-    @method findMany
-    @param {DS.Store} store
-    @param {DS.Model} type
-    @param {Array} ids
-    @param {Array} snapshots
-    @return {Promise} promise
-  */
```
Comment (bmac) [inline addon/adapters/json-api.js]: The RESTAdapter has better docs for these apis and these minimal doc blocks in the JSONAPIAdapter are preventing the better docs from being rendered. They will still be listed as public on the doc page.

**Pull request merged**